src/main/java/org/folio/rest/impl/OrdersImpl.java

```diff
@@ -80,34 +85,20 @@ public void postOrdersCompositeOrders(String lang, CompositePurchaseOrder compPO
     // First validate content of the PO and proceed only if all is okay
     helper
       .validateOrder(compPO)
-      .thenAccept(isValid -> {
+      .thenCompose(isValid -> {
         if (isValid) {
           logger.info("Creating PO and POLines...");
-          helper.createPurchaseOrder(compPO)
+          return helper.createPurchaseOrder(compPO)
             .thenAccept(withIds -> {
-              logger.info("Applying Funds...");
-              helper.applyFunds(withIds)
-                .thenAccept(withFunds -> {
-                  logger.info("Successfully Placed Order: " + JsonObject.mapFrom(withFunds).encodePrettily());
+                  logger.info("Successfully Placed Order: " + JsonObject.mapFrom(withIds).encodePrettily());
                   helper.closeHttpClient();
-                  Response response = PostOrdersCompositeOrdersResponse.respond201WithApplicationJson(withFunds,
+                  Response response = PostOrdersCompositeOrdersResponse.respond201WithApplicationJson(withIds,
                       PostOrdersCompositeOrdersResponse.headersFor201()
-                        .withLocation(String.format(ORDERS_LOCATION_PREFIX, withFunds.getId())));
+                        .withLocation(String.format(ORDERS_LOCATION_PREFIX, withIds.getId())));
                   asyncResultHandler.handle(succeededFuture(response));
-                })
-                .exceptionally(t -> {
-                  logger.error("Failure happened applying funds to created order");
-                  asyncResultHandler.handle(succeededFuture(helper.buildErrorResponse(t)));
-                  return null;
-                });
-            })
-            .exceptionally(t -> {
```

Comment:
I think one `exceptionally` block should still remain to handle any issues which might happen in `createPurchaseOrder`

Reply:
I see, it is going to be handled by outer `exceptionally`